tests/rpc/spec/location_spec.js

```diff
@@ -0,0 +1,104 @@
+describe('Location', function(){
+
+    function handleEvent(name, handler) {
+        channel.handleEvent(name, handler);
+            handlersToClean.push({
+            name: name,
+            handler: handler
+        });
+    };
+
+    beforeEach(function(done) {
+        channel.onReady(function() {
+             // Reset map and event counter.
+            channel.resetState(function() {
+                eventCounter = 0;
+                // Get default position
+                channel.getMapPosition(function(data) {
+                    defaultPosition = data;
+                    // Delay between tests
+                    setTimeout(function() {
+                        done();
+                    }, 0);
+                });
+             });
+        });
+    });
+
+    afterEach(function() {
+        // Spy callback.
+        expect(eventCounter).toEqual(1, "Event count does not match");
+        // Reset event handlers.
+        while (handlersToClean.length) {
+            var item = handlersToClean.shift();
+            channel.unregisterEventHandler(item.name, item.handler);
+        };
+    });
+
+    // # Requests: GetUserLocationRequest
+    it("Get users location", function(done) {
+
+        handleEvent('UserLocationEvent', function(data) {
+            channel.log('UserLocationEvent triggered:', data);
+            expect(data.lat).toBeDefined("latitude not found.");
+            expect(data.lon).toBeDefined("longitude not found.");
+            eventCounter++;
+            done();
+        });
+
+        channel.postRequest('MyLocationPlugin.GetUserLocationRequest', [true]);
+        channel.log('GetUserLocationRequest request done.');
+    });
+
+    // # Requests: SearchRequest
+    it("Search request", function(done) {
+
+        handleEvent('SearchResultEvent', function(data) {
+            channel.log('SearchResultEvent:', data);
+            expect(data.success).toBe(true);
+            expect(data.result.locations[0].region).toBe(searchCriteria);
+            eventCounter++;
+            done();
+        });
+
+        channel.postRequest('SearchRequest', [searchCriteria]);
+        channel.log('SearchRequest:', searchCriteria);
+    });
+
+    // # Requests: GetRouteRequest
+    all("Get route requests",[
+
+       "BICYCLE",
+       "WALK",
+       "TRANSIT,WALK"
+       ],
+        function(routeRequest, done) {
+
+            var myRouteRequest = {
+              "fromlat": defaultPosition.centerY,
+              "fromlon": defaultPosition.centerX,
+              "srs": defaultPosition.srsName,
+              "tolat": defaultPosition.centerY+1000,
+              "tolon": defaultPosition.centerX+1000,

```

Comment:
These probably won't work in a non-metric projection.

Reply:
There might be a problem with routing rpc in demo.oskari.org maps, since this didn't work with +10 either. Regardless, it's true that if the test value would be +10 it would work in both meters and degrees.
"The valid range of latitude in degrees is -90 and +90 for the southern and northern hemisphere respectively. Longitude is in the range -180 and +180 specifying coordinates west and east of the Prime Meridian, respectively."